airmozilla/popcorn/static/popcorn/js/editor.js

```diff
@@ -1,2 +1,27 @@
+function getSlug() {
+  var pathname = window.location.pathname,
+      re = /\/popcorn\/edit\/(.+)\//,
+      match = pathname.match(re);
+
+  return match[1];
+}
+
+PopcornEditor.listen(PopcornEditor.events.loaded, function () {
+  $.ajax({
+    method: 'GET',
+    url: '/popcorn/data/',
+    contentType: 'application/json',
+    data: {
+      slug: getSlug()
+    }
+  })
+  .done(function (response) {
+    if(response.exists) {
+      PopcornEditor.loadInfo(response.data);
+    } else {
+      PopcornEditor.createTemplate(response.data);
+    }
+  });
```

Comment:
This lacks a failure resolver. If you can't think of one at least start with something like this:
```javascript
.fail(function() {
    console.warn("Unable to load popcorn data :(");
    console.error.apply(console, arguments);
})
```
Or, if you don't have time, add an `alert()` because otherwise only people who open the web console will be informed.